core/trino-main/src/main/java/io/trino/execution/CreateViewTask.java

```diff
@@ -81,13 +82,21 @@ public ListenableFuture<Void> execute(
         if (metadata.isMaterializedView(session, name)) {
             throw semanticException(TABLE_ALREADY_EXISTS, statement, "Materialized view already exists: '%s'", name);
         }
-        if (metadata.isView(session, name)) {
-            if (!statement.isReplace()) {
-                throw semanticException(TABLE_ALREADY_EXISTS, statement, "View already exists: '%s'", name);
+
+        try {
+            if (metadata.isView(session, name)) {
+                if (!statement.isReplace()) {
+                    throw semanticException(TABLE_ALREADY_EXISTS, statement, "View already exists: '%s'", name);
+                }
+            }
+            else if (metadata.getTableHandle(session, name).isPresent()) {
+                throw semanticException(TABLE_ALREADY_EXISTS, statement, "Table already exists: '%s'", name);
             }
         }
-        else if (metadata.getTableHandle(session, name).isPresent()) {
-            throw semanticException(TABLE_ALREADY_EXISTS, statement, "Table already exists: '%s'", name);
+        catch (UnsupportedViewDialectException exception) {
+            if (!statement.isReplace()) {
+                throw semanticException(TABLE_ALREADY_EXISTS, statement, "View %s already exists with dialect '%s'", name, exception.getDialect());
```

Comment:
The one case we need to handle for the new exception is for `CREATE OR REPLACE`. We should allow REPLACE on views for non-trino dialects assuming the https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/view/ViewProperties.java#L29 property is enabled. The validation for that is done by Iceberg catalogs.

Reply:
I think the core problem is Trino engine metadata is conflating `isView(name)` to `getView(name).isPresent()`.  If instead we passed that through to the `ConnectorMetadata` then Iceberg could return "yes this is a view".  If you actually tried to use it you would get an error, but that seems ok to me. @electrum what do you think?

Reply:
This makes sense to me. We should add an `isView()` method to `ConnectorMetadata` and remove the default implementation from `Metadata`:
```java
default Optional<ConnectorViewDefinition> isView(ConnectorSession session, SchemaTableName viewName)
{
    return getView(session, viewName).isPresent();
}
```
The Iceberg connector can implement this method. With that change, we shouldn't need any other changes to the engine (no error code or exception). The Iceberg connector can throw `NOT_SUPPORTED` when `getView()` is called for a non-Trino view.